src/lua/zencode_sd_jwt.lua

```diff
@@ -27,110 +27,94 @@ local function import_url_f(obj)
 end
 
 local URLS_METADATA = {
-    "issuer",
     "credential_issuer",
-    "authorization_endpoint",
-    "token_endpoint",
-    "pushed_authorization_request_endpoint",
-    "credential_endpoint",
-    "jwks_uri",
-}
-
-local DICTS_METADATA = {
-    "grant_types_supported",
-    "code_challenge_methods_supported",
-    "response_modes_supported",
-    "response_types_supported",
-    "scopes_supported",
-    "subject_types_supported",
-    "token_endpoint_auth_methods_supported",
-    "id_token_signing_alg_values_supported",
-    "request_object_signing_alg_values_supported",
-    "claim_types_supported",
-}
-
-local BOOLS_METADATA = {
-    "claims_parameter_supported",
-    "authorization_response_iss_parameter_supported",
-    "request_parameter_supported",
-    "request_uri_parameter_supported",
+    "credential_endpoint"
 }
+--"authorization_servers",
+
+-- local DICTS_METADATA = {
+--     "grant_types_supported",
+--     "code_challenge_methods_supported",
+--     "response_modes_supported",
+--     "response_types_supported",
+--     "scopes_supported",
+--     "subject_types_supported",
+--     "token_endpoint_auth_methods_supported",
+--     "id_token_signing_alg_values_supported",
+--     "request_object_signing_alg_values_supported",
+--     "claim_types_supported",
+-- }
+
+-- local BOOLS_METADATA = {
+--     "claims_parameter_supported",
+--     "authorization_response_iss_parameter_supported",
+--     "request_parameter_supported",
+--     "request_uri_parameter_supported",
+-- }
 
 local function check_display(display)
     return display.name and display.locale
 end
 
 local function import_supported_selective_disclosure(obj)
-    local check_support = function(what, needed)
+    local check_support = function(obj, what, needed)
         found = false
-        for i=1,#obj[what] do
-            if obj[what][i] == needed then
-                found = true
-                break
+        if(type(obj[what]) == 'table') then
+            for k,v in pairs(obj[what]) do
+               if v == needed[k] then
+                    found = true
+                    break
+               end 
             end
+        else 
+            found = obj[what] == needed
         end
-        assert(found, needed .. " not supported in " .. what)
+        assert(found, " not supported in " .. what)
```

Comment:
since this is not inside a zencode statement function, the error(..., 2) call should be used instead with a check, so that the line returned by the error will be pointing to the caller of this function.